src/wms/src/main/java/org/geoserver/wms/legendgraphic/TalliedQueue.java

```diff
@@ -0,0 +1,58 @@
+/* (c) 2014 - 2022 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2022 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms.legendgraphic;
+
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class TalliedQueue<T> implements ImageSizeComputable {
```

Comment:
This seems quite the heavy weight solution; how does a normal GetMap request respect time constraint?

Reply:
GetMap has a simpler computation, width*height*number of feature type styles* bands.
Here due to the various layouting options and symbology sizes, it's difficult to predict how big the legend will become.

I don't see where the "heavy weight" is, the code is already using collections to keep track of the various sub-images that it will have to lay out later, these new classes do the same keeping track of how much memory has been used in the process. Please elaborate on how you'd do the same in a simpler way.

Reply:
Okay cool; I just wanted to make sure that introducing a new class ``TalliedQueue`` was not duplicating work already done for GetMap.